### PR TITLE
Add HTTP status codes 420, 421, 451, 508

### PR DESCRIPTION
**Pull request by burnersk**

### CODE REVIEW

**Comment by genio** [comment]
While I don't necessarily think it's wrong, some of the codes you added aren't really official. The official list of status codes is here: [IANA Status Codes](http://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml)

File: lib/HTTP/Status.pm [hunk 1]
```diff
@@ -55,6 +55,8 @@ my %StatusCode = (
     416 => 'Request Range Not Satisfiable',
     417 => 'Expectation Failed',
     418 => 'I\'m a teapot',		      # RFC 2324
+    420 => 'Policy Not Fulfilled',
+    421 => 'Too Many Connections From Your IP Address',
```
Comment (oalders) [inline lib/HTTP/Status.pm]: RFC 7540 seems to indicate that `421` is "Misdirected request". https://tools.ietf.org/html/rfc7540#page-66

**Comment by pangyre** [comment]
While on the matter of new status codes, I put together a list and helper script to diff HTTP::Status codes from those that Wikipedia lists. There are at least 22 differences today. I understand there are concerns about formality of codes and such so I'm not boostering for adding everything, just giving information.

Gist of tool: https://gist.github.com/pangyre/39226b024e9e536d054fddd8debd33d1

Current output (better looking in monospace)-

Nothing known about 103 [MULTIPLE DEFINITIONS]
                        -> Checkpoint
                        -> Early Hints
Nothing known about 226 -> IM Used (RFC 3229)
Nothing known about 306 -> Switch Proxy
Nothing known about 420 [MULTIPLE DEFINITIONS]
                        -> Method Failure (Spring Framework)
                        -> Enhance Your Calm (Twitter)
Nothing known about 421 -> Misdirected Request (RFC 7540)
Nothing known about 440 -> Login Time-out
Nothing known about 444 -> No Response
Nothing known about 450 -> Blocked by Windows Parental Controls (Microsoft)
Nothing known about 451 [MULTIPLE DEFINITIONS]
                        -> Unavailable For Legal Reasons (RFC 7725)
                        -> Redirect
Nothing known about 495 -> SSL Certificate Error
Nothing known about 496 -> SSL Certificate Required
Nothing known about 497 -> HTTP Request Sent to HTTPS Port
Nothing known about 498 -> Invalid Token (Esri)
Nothing known about 499 [MULTIPLE DEFINITIONS]
                        -> Token Required (Esri)
                        -> Client Closed Request
Nothing known about 508 -> Loop Detected (WebDAV; RFC 5842)
Nothing known about 520 -> Unknown Error
Nothing known about 521 -> Web Server Is Down
Nothing known about 522 -> Connection Timed Out
Nothing known about 523 -> Origin Is Unreachable
Nothing known about 524 -> A Timeout Occurred
Nothing known about 525 -> SSL Handshake Failed
Nothing known about 526 -> Invalid SSL Certificate
Nothing known about 527 -> Railgun Error
Nothing known about 530 -> Site is frozen
Nothing known about 598 -> (Informal convention) Network read timeout error
Nothing known about 599 -> (Informal convention) Network connect timeout error

**Comment by vanHoesel** [comment]
#100(Update status codes to official IANA list) obsoletes this merge-request - and sticks to the official IANA list, declining `420`, which has multiple meanings

**Comment by vanHoesel** [comment]
this can be closed, since https://github.com/libwww-perl/HTTP-Message/pull/100 has been merged, disregarding the numbers requested here

**Comment by oalders** [comment]
Thanks very much for taking the time to put this together @burnersk.  I'll close for the reasons pointed out by @vanHoesel.

**Pull request closed without merging**